src/commands/types.d.ts

```diff
@@ -7,7 +7,7 @@ import StateConfig from '../utils/state-config.js'
 type $TSFixMe = any;
 
 export type NetlifySite = {
-  root?: string
+  root: string
```

Comment:
I don't think the root is always defined; I believe we need to cope with scenarios where it's undefined. I would propose doing a type guard on the usage instead of making the type required all the time

Reply:
addressed in https://github.com/netlify/cli/pull/6257/commits/c4136e5b39f746e472ed9ae1c95193a871c1dfd2